proxy/src/main/java/com/wavefront/agent/preprocessor/ReportPointAddTagIfNotExistsTransformer.java

```diff
@@ -11,17 +13,25 @@
  */
 public class ReportPointAddTagIfNotExistsTransformer extends ReportPointAddTagTransformer {
 
+  @Nullable
+  private final Map<String, Object> v2Predicate;
+
   public ReportPointAddTagIfNotExistsTransformer(final String tag,
                                                  final String value,
+                                                 @Nullable final Map<String, Object> v2Predicate,
```

Comment:
I think this could be another performance issue, as you are basically parsing your predicate for every point. A better approach would be to parse the entire predicate tree from your Map<String, Object> into Predicate<T> and pass that as an argument. That would also allow you to cache compiled regex patterns, etc.